### PR TITLE
fix `Don't update urlResolver if page already contains `base` tag` (close #1349)

### LINKED ISSUE #1349: Don't update urlResolver if page already contains `base` tag
We should not update `urlResolver` base path if page already contains `base` tag.
We should to emulate native browser behavior and consider only first one.

To reproduce the issue need to open  https://miherlosev.github.io/issue-pages/hammerhead/1349/folder/page.html url in hammerhead playground.
In console will displayed a resolved url
Without proxy: https://miherlosev.github.io/0.js
With proxy: https://miherlosev.github.io/issue-pages/hammerhead/1349/folder/0.js

Related to https://github.com/DevExpress/testcafe/issues/1912

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev @churkin

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: test/client/fixtures/utils/url-test.js [hunk 1]
```diff
@@ -557,3 +559,95 @@ test('resolving a url in a tag that is written along with a "base" tag (GH-644)'
...
+    var storedProcessElement = nodeSandbox._processElement;
+    var nativeIframe         = nativeMethods.createElement.call(document, 'iframe');
+    var proxiedIframe        = null;
+
+    nodeSandbox._processElement = function (el) {
+        if (el !== nativeIframe)
+            storedProcessElement.call(nodeSandbox, el);
+    };
+
+    function compareWithNative (name, fn) {
```
Comment (miherlosev) [inline test/client/fixtures/utils/url-test.js]: `function checkTestCase(name, fn) {`

File: test/client/fixtures/utils/url-test.js [hunk 2]
```diff
@@ -557,3 +559,95 @@ test('resolving a url in a tag that is written along with a "base" tag (GH-644)'
 
     document.body.removeChild(iframe);
 });
+
+test('compare with native behavior', function () {
```
Comment (miherlosev) [inline test/client/fixtures/utils/url-test.js]: `test('only first base tag should be affected')`

File: src/client/sandbox/node/element.js [hunk 3]
```diff
@@ -543,6 +546,11 @@ export default class ElementSandbox extends SandboxBase {
         return el.parentNode && domUtils.isShadowUIElement(el.parentNode) || ShadowUI.containsShadowUIClassPostfix(el);
     }
 
+    _isNeedToUpdateUrlResolver (el) {
+        return domUtils.isElementInDocument(el, this.document) &&
```
Comment (miherlosev) [inline src/client/sandbox/node/element.js]: Also add tests for `documentFragment` and for `setProperty(container, 'innerHTML', value) cases

**Comment by LavrovArtem** [comment]
ping @churkin

**Pull request merged**